prometheus_sink.go

```diff
@@ -0,0 +1,51 @@
+package stats
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type prometheusSink struct {
+	counters   map[string]prometheus.Counter
+	gauges     map[string]prometheus.Gauge
+	histograms map[string]prometheus.Histogram
+}
+
+// NewLoggingSink returns a Sink that flushes stats to os.StdErr.
+func NewPrometheusSink() Sink {
+	http.Handle("/metrics", promhttp.Handler())
+	go http.ListenAndServe(":2112", nil)
```

Comment:
any reason for `":2112"` (ie vs ":9090")?

Reply:
copypasta